tools/docker/Dockerfile.ubuntu-build

```diff
@@ -0,0 +1,61 @@
+# syntax=docker/dockerfile:1
+FROM ubuntu:20.04 as builder
+
+WORKDIR /build
+
+COPY . ./
+
+# install build dependencies
+# taken from https://github.com/dragonflydb/dragonfly/blob/main/.github/workflows/release.yml#L60-L65
+RUN \
+  rm -f /etc/apt/apt.conf.d/docker-clean; \
+  echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+
+RUN \
+  --mount=type=cache,target=/var/cache/apt,sharing=locked \
```

Comment:
can you explain what you do here? I understand that it's somehow connected to apt caching but I do not understand exactly what it is. If you have a reference to a blog post or documentation this will also work.

Reply:
Theoretically, that's supposed to cache and share `/var/cache/apt` and `/var/lib/apt` between both builds, but also between runs (through the `cache-{to|from}: type=gha` parameter). But doesn't seem to work as I'd hoped.